mk/support/pkg/curl.sh

```diff
@@ -96,7 +102,7 @@ pkg_link-flags () {
             -lidn)    out `pkg link-flags libidn idn` ;;
             -lssl)    out_openssl ssl ;;
             -lcrypto) out_openssl crypto ;;
-            -ldl)     dl_libs=-ldl;; # Linking may fail if -ldl isn't last
```

Comment:
We'll need to make sure that this change doesn't cause issues on other platforms. I assume the `-ldl` was there for a reason.